source/macos/AdaptiveCards/AdaptiveCards/Utils/HostConfigUtils.swift

```diff
@@ -157,6 +157,29 @@ class TextUtils {
         }
         return content
     }
+    
+    static func addFontProperties(attributedString: NSMutableAttributedString, textProperties: ACSRichTextElementProperties, hostConfig: ACSHostConfig) -> NSMutableAttributedString {
+        let content = NSMutableAttributedString(attributedString: attributedString)
+        attributedString.enumerateAttributes(in: NSRange(location: 0, length: attributedString.length), options: .longestEffectiveRangeNotRequired, using: { attributes, range, _ in
+            if let font = attributes[.font] as? NSFont {
+                let fontWithProperties = FontUtils.getFont(for: hostConfig, with: textProperties)
+                var descriptor = font.fontDescriptor
+                
+                if descriptor.symbolicTraits.contains(.italic) {
+                    descriptor = descriptor.withFamily(fontWithProperties.familyName ?? "").withSymbolicTraits(.italic)
+                } else if descriptor.symbolicTraits.contains(.bold) {
+                    descriptor = descriptor.withFamily(fontWithProperties.familyName ?? "").withSymbolicTraits(.bold)
+                } else {
+                    descriptor = descriptor.withFamily(fontWithProperties.familyName ?? "")
+                    let weightAttributes = CTFontDescriptorCopyAttributes(fontWithProperties.fontDescriptor)
+                    descriptor = CTFontDescriptorCreateCopyWithAttributes(descriptor, weightAttributes)
+                }
+                let newFont = NSFont(descriptor: descriptor, size: fontWithProperties.pointSize)
+                content.addAttribute(.font, value: newFont as Any, range: range)
+            }
```

Comment:
```suggestion
            guard let font = attributes[.font] as? NSFont else { return }
            let fontWithProperties = FontUtils.getFont(for: hostConfig, with: textProperties)
            var descriptor = font.fontDescriptor.withFamily(fontWithProperties.familyName ?? "")

            if descriptor.symbolicTraits.contains(.italic) {
                descriptor = descriptor.withSymbolicTraits(.italic)
            } else if descriptor.symbolicTraits.contains(.bold) {
                descriptor = descriptor.withSymbolicTraits(.bold)
            } else {
                let weightAttributes = CTFontDescriptorCopyAttributes(fontWithProperties.fontDescriptor)
                descriptor = CTFontDescriptorCreateCopyWithAttributes(descriptor, weightAttributes)
            }
            let newFont = NSFont(descriptor: descriptor, size: fontWithProperties.pointSize)
            content.addAttribute(.font, value: newFont as Any, range: range)
```
Maybe this will do?

Reply:
Akaash all the descriptor properties have to be added at once else certain things do not work as priority changes(Markdown stops working in this case)